contentctl/actions/test.py

```diff
@@ -45,16 +45,6 @@ class TestInputDto:
 
 class Test:
 
-    def filter_tests(self, input_dto: TestInputDto) -> TestInputDto:
-        if not input_dto.config.enable_integration_testing:
```

Comment:
The app is available during model construction, but I think we should make the whole config available during Detection construction. Here is the current line: https://github.com/splunk/contentctl/blob/f66c60f6fe6756e1418f0a0890c2689ae871a152/contentctl/input/director.py#L197

Reply:
After some thoughts, I have added the filtering back to  the original location. We can't/shouldn't do it at Detection Object creation because at that time we  _must not_ require a `test_common` configuration (test object creation happens during `validate` and `build` operations, not just `test` operations.  As such, we can't decide whether or not integration tests should be skipped.

We can think about making the test_common object a more robust pydantic class in the future (where its own `model_post_init` function or similar manipulates tests or things like that) but I think what we had already is a fair approach for now.